provider/pkg/provider/provider.go

```diff
@@ -550,13 +551,39 @@ func (k *azureNativeProvider) getDefaultLocation(ctx context.Context, olds, news
 	return result(v)
 }
 
-func (k *azureNativeProvider) getDefaultName(urn string, randomSeed []byte, strategy resources.AutoNameKind,
-	key resource.PropertyKey, olds resource.PropertyMap) (resource.PropertyValue, bool) {
+func (k *azureNativeProvider) getDefaultName(urn string, randomSeed []byte,
+	autonaming *rpc.CheckRequest_AutonamingOptions, strategy resources.AutoNameKind, key resource.PropertyKey,
+	olds resource.PropertyMap) (nameValue *resource.PropertyValue, randomlyNamed bool, ok bool) {
 	if v, ok := olds[key]; ok {
 		if vf, ok := olds[createBeforeDeleteFlag]; ok && vf.IsBool() {
-			return v, vf.BoolValue()
+			return &v, vf.BoolValue(), true
+		}
+		return &v, false, true
+	}
+
+	result := func(s string) *resource.PropertyValue {
+		p := resource.NewStringProperty(s)
+		return &p
+	}
+
+	if autonaming != nil {
+		switch autonaming.Mode {
+		case rpc.CheckRequest_AutonamingOptions_DISABLE:
+			// Do not return any value. If the property is required, Check will fail
+			// with a missing required property error.
+			return nil, false, false
+		case rpc.CheckRequest_AutonamingOptions_ENFORCE:
+			return result(autonaming.ProposedName), true, true
+		case rpc.CheckRequest_AutonamingOptions_PROPOSE:
+			switch strategy {
+			case resources.AutoNameRandom:
+				fallthrough
+			case resources.AutoNameCopy:
+				return result(autonaming.ProposedName), true, true
```

Comment:
Not seen `fallthrough` in Go code before .. assume this is the same as:

```suggestion
			case resources.AutoNameRandom, resources.AutoNameCopy
				return result(autonaming.ProposedName), true, true
```